### PR TITLE
Add support for setting isZoneRedundant property of Cosmos account region

### PR DESCRIPTION
**Pull request by shurd**
Introduce a new format for specifying location details of a Cosmos account. This new format allows users to specify the boolean property "isZoneRedundant" when adding a region. This change also deprecates the old method of specifying locations (regionName=failoverPriority).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/_params.py [hunk 1]
```diff
@@ -27,7 +30,7 @@ def load_arguments(self, _):
...
             c.argument('account_name', completer=None)
             c.ignore('resource_group_location')
-            c.argument('locations', nargs='+', validator=validate_locations, help="space-separated locations in 'regionName=failoverPriority' format. E.g eastus=0 westus=1. Failover priority values are 0 for write regions and greater than 0 for read regions. A failover priority value must be unique and less than the total number of regions. Default: single region account in the location of the specified resource group.")
+            c.argument('locations', nargs='+', action=CreateLocation, help="space-separated key-value pairs for properties of a location formatted [KEY=VALUE ...]. Required keys are 'regionname' and 'failoverpriority'. Optional boolean key is 'iszoneredundant'. E.g. regionname=eastus failoverpriority=0 iszoneredundant=false. Failover priority values are 0 for write regions and greater than 0 for read regions. A failover priority value must be unique and less than the total number of regions. Default: single region account in the location of the specified resource group. Multiple locations can be specified by using more than one `--locations` argument.")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/_params.py]: Everything beyond the first sentence should be part of a `long_summary` entry in the `help.py` file. See `az monitor autoscale create` for an example.

Comment (tjprescott) [inline src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/_params.py]: Also, do include an example for this command and its updated usage in your `help.py` file.

File: src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/actions.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                elif _key == "iszoneredundant":
+                    _is_zr = kvp[1].lower() == "true"
+                else:
+                    raise CLIError('usage error: --locations [KEY=VALUE ...]')
+            namespace.locations.append(
+                Location(location_name=_name, 
+                         failover_priority=_failover, 
+                         is_zone_redundant=_is_zr))
+        else:
+            logger.warning('DEPRECATION WARNING: The regionName=failoverPriority method of specifying locations is deprecated. Use --locations [KEY=VALUE ...] to specify the regionName, failoverPriority, and isZoneRedundant properties of the location. Multiple locations can be specified by including more than one --locations argument.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/actions.py]: Consider here that if someone writes `--location eastus=1 --location westus=2` then they will see this warning twice. You might want to track a private variable on the namespace to track whether the warning has been displayed once and not display it again.  If you are okay with bugging the user to move to the new format, then this is fine. :)

File: src/command_modules/azure-cli-cosmosdb/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.2.11
+++++++
+* Added support for setting "isZoneRedundant" property of location. Introduced new format for specifying location details of an account and deprecated the old format.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-cosmosdb/HISTORY.rst]: Better to be more specific here. `* cosmsodb create/update: Added new format for --location to allow setting "isZoneRedundant" property. Deprecated old format.`

File: src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/custom.py [hunk 4]
```diff
@@ -104,10 +109,15 @@ def cli_cosmosdb_update(client,
...
                         virtual_network_rules=None,
-                        enable_multiple_write_locations=None):
+                        enable_multiple_write_locations=None,
+                        deprecated_location=None):
     """Update an existing Azure Cosmos DB database account. """
     existing = client.get(resource_group_name, account_name)
 
+    # pylint: disable=line-too-long
+    if deprecated_location:
+        logger.warning('DEPRECATION WARNING: The regionName=failoverPriority method of specifying locations is deprecated. Use --locations [KEY=VALUE ...] to specify the regionName, failoverPriority, and isZoneRedundant properties of the location. Multiple locations can be specified by including more than one --locations argument.')
```
[the remark this answers is not in the document]
Comment (shurd) [reply src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/custom.py]: I will remove it. I was trying to find how to deprecate a format as the --locations parameter is still valid, so I copied what I found in `az bot publish`.

**Comment by shurd** [comment]
Is there a way to re-trigger the build? The Travis-CI timed out after my most recent commit, and I do not think the text changes I made in that change would cause it to go from passing to failing.

**Pull request merged**